### PR TITLE
Add version to help display

### PR DESCRIPTION
**Pull request by BrennanConroy**
#1336

### CODE REVIEW

**Comment by davidfowl** [comment]
Screen shots on windows and on mono plz.

**Comment by BrennanConroy** [comment]
screen shot 2015-03-19 at 2 57 40 pm

**Comment by BrennanConroy** [comment]
dnxversion
@davidfowl

File: src/Microsoft.Framework.CommandLineUtils/CommandLine/CommandLineApplication.cs [hunk 1]
```diff
@@ -322,6 +322,7 @@ public void ShowHelp(string commandName = null)
                 headerBuilder.Insert(6, string.Format(" {0}", cmd.Name));
             }
 
+            var version = string.Format("Microsoft .NET Execution environment v{0}{1}{1}", VersionGetter(), Environment.NewLine);
```
Comment (ChengTian) [inline src/Microsoft.Framework.CommandLineUtils/CommandLine/CommandLineApplication.cs]: `CommandLineApplication` is shared by a lot of command-line tools. We shouldn't show this string when you check help info of `kpm` and `dnx`.

I suggest we add a `FullName` property, which can be set by users of `CommandLineApplication`

Comment (davidfowl) [inline src/Microsoft.Framework.CommandLineUtils/CommandLine/CommandLineApplication.cs]: +1

**Comment by BrennanConroy** [comment]
dnx
dnu

**Comment by ChengTian** [comment]
Very nice! :+1:  :shipit:

**Comment by davidfowl** [comment]
Pedantic but relevant:

- Don't call GetVersion on the startup path. Make it lazy (it already is)

**Comment by ChengTian** [comment]
Good point. Yeah, we should always make sure we don't ruin @davidfowl 's efforts on start-up optimization :smile:

**Comment by BrennanConroy** [comment]
My bad! Fixed

File: src/Microsoft.Framework.PackageManager/Program.cs [hunk 2]
```diff
@@ -36,6 +36,7 @@ public int Main(string[] args)
         {
             var app = new CommandLineApplication();
             app.Name = "dnu";
+            app.FullName = "Microsoft .NET Utility";
```
Comment (davidfowl) [inline src/Microsoft.Framework.PackageManager/Program.cs]: Check with @danroth27 on this name.

**Comment by BrennanConroy** [comment]
He wants it to be "Microsoft .NET Development Utility"

File: src/Microsoft.Framework.PackageManager/Program.cs (hunk 2 above)
Comment (BrennanConroy) [inline src/Microsoft.Framework.PackageManager/Program.cs]: He wants it to be "Microsoft .NET Development Utility"

**Comment by davidfowl** [comment]
:shipit:

**Pull request closed without merging**